DirectEmployers/MyJobs
# [PD-3487] Add PartnerAssessment handling to reporting

## What's this PR do?
This PR is the combination of a few different tickets. It does the following:

1) Adds Score, Review to the Communication Reports
2) Adds a new report category for PRM, "Partner Assessments"
3) Adds a datasource for exporting Partner Assessment Reports
4) Adds a simple mechanism to waffle-gate these new reporting changes.

## Where should the reviewer start?
Django Admin
* Navigate to the bottom of the admin page, to "Waffle"
* Click the "Add" button next to "Switches" (NOT flags)
* Add a new switch called "partner assessment"

## How should this be manually tested?
Reporting App
* Select PRM
* Select Communication Record (default selection)
* Add score filter.
* Export a report
* Check that it contains score/review fields
* Select "Partner Assessment" from Report Category
* Ensure the form/export works properly.

## Any background context you want to provide?
Due to difficulties passing the request object into certain functions, I spoke with Jason Sole and we decided that we did not need user-specific waffle functionality for this app. Instead of using flags (user specific), we are using switches (on or off for all).

Also, there is a conspicuous lack of tests for the new datasource. I have created a spin-off ticket [PD-3633](https://directemployersfoundation.atlassian.net/browse/PD-3633) for that and am currently working on those tests.
